datamade/bga-payroll
# Add custom text for the donate banner

For issue #514.

Also fixed something in the readme.

## Notes
I'm using the jinja `safe` filter in combination with the `bleach` library for sanitizing the user-input html. `bleach` lets us whitelist any html tags we want to allow. The `safe` filter then parses the html string into html. Conversation in issue #514 has more details.

## Demo
See this screenshot, with the marked-up text and the unwanted <script> tag. If the script tag was whitelisted with bleach, then that code would execute in the console instead of being added to the page. See also the source code screenshot, where the script tag is sanitized.
<img width="1318" alt="Screen Shot 2021-08-18 at 3 15 26 PM" src="https://user-images.githubusercontent.com/38969506/130245821-851912a8-3bc0-4d44-aeb1-0d1f6c06360c.png">
<img width="645" alt="Screen Shot 2021-08-20 at 9 07 03 AM" src="https://user-images.githubusercontent.com/38969506/130245858-913f8b46-b99d-4775-a1fa-7c1a6591edc7.png">


## Testing
- checkout `frontend-changes` branch locally (needs better name - sorry!)
- in the admin interface, 
  - add the `PAYROLL_SHOW_DONATE_BANNER` setting and set it to true
  - add the `DONATE_MESSAGE` and input some html with `<p>` and `<strong>` tags. It should show up.
